community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -779,6 +789,25 @@ public int getDegree( RelationshipType type, Direction direction )
         }
     }
 
+    //Part of NodeValue API
+
+    @Override
+    public TextArray labels()
+    {
+        ArrayList<String> ls = new ArrayList<>();
+        for ( Label label : getLabels() )
+        {
+            ls.add( label.name() );
+        }
+        return Values.stringArray( ls.toArray( new String[ls.size()] ) );
+    }
+
+    @Override
+    public MapValue properties()
+    {
+        return ValueUtils.asMapValue( getAllProperties() );
```

Comment:
This seems like the wrong way around, going up the abstraction hierarchy only to go back down.